File: CaCompteKit/Sources/Domain/Stats/Badge.swift
/// Doc 06 « Badges » — décernés en fin de partie, un par joueur au maximum, purement décoratifs.
public struct Badge: Sendable, Equatable {
    public enum Kind: String, Sendable, Equatable {
        case winner
        case metronome
        case rollercoaster
        case comeback
        case kamikaze
        case unshakeable
        case photoFinish
        case sniper
        case boulet
        case landslide
    }

    public let kind: Kind
    public let participantID: Participant.ID

    public init(kind: Kind, participantID: Participant.ID) {
        self.kind = kind
        self.participantID = participantID
    }
}

File: android/domain/src/main/kotlin/com/cacompte/domain/stats/Badge.kt
package com.cacompte.domain.stats

import java.util.UUID

/** Miroir de `Badge.swift` — pas de sérialisation (résultat de calcul en mémoire, jamais
 * transmis ni persisté, comme `ValidationResult`). */
data class Badge(
    val kind: Kind,
    val participantID: UUID,
) {
    enum class Kind {
        Winner,
        Metronome,
        Rollercoaster,
        Comeback,
        Kamikaze,
        Unshakeable,
        PhotoFinish,
        Sniper,
        Boulet,
        Landslide,
    }
}
